src/inbox.cpp

```diff
@@ -31,23 +32,25 @@ ReturnValue Inbox::queryAdd(int32_t, const Thing& thing, uint32_t, uint32_t flag
 	return RETURNVALUE_NOERROR;
 }
 
-void Inbox::postAddNotification(Thing* thing, const Cylinder* oldParent, int32_t index, cylinderlink_t)
+void Inbox::postAddNotification(std::shared_ptr<Thing> thing, std::shared_ptr<const Thing> oldParent, int32_t index,
+                                cylinderlink_t)
 {
-	Cylinder* parent = getParent();
+	auto parent = getParent();
```

Comment:
Could this be `const auto`
```suggestion
	const auto parent = getParent();
```